apps/prairielearn/src/lib/file-store.ts

```diff
@@ -120,28 +132,28 @@ export async function deleteFile(file_id, authn_user_id) {
 /**
  * Option of returning a stream instead of a file
  *
- * @param {number | string} file_id - The file to get.
- * @return {Promise<(import('stream'))>} - Requested file stream.
+ * @param file_id - The file to get.
+ * @return Requested file stream.
  */
-export async function getStream(file_id) {
+export async function getStream(file_id: number | string): Promise<stream> {
   debug(`getStream(): file_id=${file_id}`);
   const file = await getFile(file_id, 'stream');
-  return file.contents;
+  return file.contents as stream /* Strange conversion */;
```

Comment:
You could probably fix this by improving the types of `getFile` using overloads: https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads. The types can then be aware of the return type based on `data_type`.

Reply:
I'm not super sure how to resolve this. I tried using an overload, but even if the type is correct it doesn't seem to like it.

```
Type 'StreamingBlobTypes' is not assignable to type 'internal'.
  Type 'ReadableStream<any>' is missing the following properties from type 'internal': pipe, compose, addListener, on, and 13 more.
  ```

Reply:
The changes to our AWS lib from #11011 might help here. Let's wait for that to land and see what happens.

Reply:
Oops, I had mistyped ReadStream.

Reply:
This is not resolved; `getFromS3` still needs overloads for buffer vs. stream.